fcrepo-jms/src/test/java/org/fcrepo/integration/jms/observer/JmsIT.java

```diff
@@ -172,24 +167,24 @@ public void testMetadataEvents() throws RepositoryException {
             final String sparql1 = "insert data { <> <http://foo.com/prop> \"foo\" . }";
             resource1.updateProperties(subjects, sparql1, resource1.getTriples(subjects, PROPERTIES));
             session.save();
-            awaitMessageOrFail(testMeta, PROP_ADDED_EVENT_TYPE, "http://foo.com/prop");
+            awaitMessageOrFail(testMeta, RESOURCE_MODIFICATION.getType(), REPOSITORY_NAMESPACE + "Container");
 
             final FedoraResource resource2 = containerService.findOrCreate(session, testMeta);
             final String sparql2 = " delete { <> <http://foo.com/prop> \"foo\" . } "
                 + "insert { <> <http://foo.com/prop> \"bar\" . } where {}";
             resource2.updateProperties(subjects, sparql2, resource2.getTriples(subjects, PROPERTIES));
             session.save();
-            awaitMessageOrFail(testMeta, PROP_CHANGED_EVENT_TYPE, "http://foo.com/prop");
+            awaitMessageOrFail(testMeta, RESOURCE_MODIFICATION.getType(), REPOSITORY_NAMESPACE + "Resource");
         } finally {
             session.logout();
         }
     }
 
-    private void awaitMessageOrFail(final String id, final String eventType, final String property) {
-        await().pollInterval(ONE_SECOND).until(() -> messages.stream().anyMatch(msg -> {
+    private void awaitMessageOrFail(final String id, final String eventType, final String type) {
+        await().pollInterval(ONE_HUNDRED_MILLISECONDS).until(() -> messages.stream().anyMatch(msg -> {
             try {
                 return getPath(msg).equals(id) && getEventTypes(msg).contains(eventType)
-                        && (property == null || getProperties(msg).contains(property));
+                        && (type == null || getResourceTypes(msg).contains(type));
```

Comment:
No, all strings contain the empty string.



> On Jun 17, 2016, at 4:40 PM, Aaron Coburn <notifications@github.com> wrote:
> 
> In fcrepo-jms/src/test/java/org/fcrepo/integration/jms/observer/JmsIT.java:
> 
> >              try {
> >                  return getPath(msg).equals(id) && getEventTypes(msg).contains(eventType)
> > -                        && (property == null || getProperties(msg).contains(property));
> > +                        && (type == null || getResourceTypes(msg).contains(type));
> Here we are distinguishing between an input parameter that is either null (in that case, just evaluate to true) or, in the case of a non-null value, check that it appears in the Set of resource types. With nullToEmpty, I suppose we could make the line into:
> 
> && (nullToEmpty(type).isEmpty() || getResourceTypes(msg).contains(type));
> but I don't think that's any more clear.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>